### DISCUSSION TITLE
`undefined` requires Angular-CLI to run

### DISCUSSION THREAD

**Discussion by akaustav** [coveo/cli#986] [Q&A] [answered]
Hi,
In my Windows 10 machine, after installing the `@coveo/cli@1.37.1` globally using `npm` (via the `npm -g i @coveo/cli` command) and logging in (via the [`coveo auth:login`](https://github.com/coveo/cli/tree/master/packages/cli/core#coveo-authlogin) command), I attempted to create a sample Angular Headless application. However, as soon as I issued the following command I was hit with an error:

```shell
C:\dev\coveo>coveo ui:create:angular coveo-test
 »   Precondition Error - Missing Bin: undefined requires Angular-CLI to run.
 »
 »   You can install the Angular-CLI by running npm i -g @angular/cli
 »
 »   Please visit https://angular.io/guide/setup-local#install-the-angular-cli for more detailed installation
 »   information.
```

I was able to deduce that I was missing a global installation of the `@angular/cli`. I was able to install it and move on with the remainder steps. However, it feels like the `undefined` in the error message should point to a specific command / package / library. Is this a known bug?

If it helps, I am running:

- **OS:** `Windows 10 Version 21H2 (OS Build 19044.2006)`
- **NodeJS:** `16.10.0`
- **NPM:** `7.24.0`
- **Coveo CLI:** `@coveo/cli/1.37.1 win32-x64 node-v16.10.0`

Ameet

**Comment by louis-bompart** [comment, marked as answer]
Hey @akaustav , thanks for the report, I reproduced your issue and this seems to be a bug indeed.
It is tracked under #987.

Thanks again! 🙌
